laser_encoders/laser_tokenizer.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# LASER  Language-Agnostic SEntence Representations
+# is a toolkit to calculate multilingual sentence embeddings
+# and to use them for document classification, bitext filtering
+# and mining
+#
+# --------------------------------------------------------
+#
+# Helper functions for tokenization
+
+import gzip
+import logging
+import re
+import sys
+from pathlib import Path
+from typing import IO, List
+
+import sentencepiece as spm
+from sacremoses import MosesDetokenizer, MosesPunctNormalizer
+
+SPACE_NORMALIZER = re.compile(r"\s+")
+
+logging.basicConfig(
+    stream=sys.stdout,
+    level=logging.INFO,
+    format="%(asctime)s | %(levelname)s | %(name)s | %(message)s",
+)
+logger = logging.getLogger("preprocess")
+
+
+class LaserTokenizer:
+    def __init__(
+        self,
+        spm_model: Path,
+        lang: str = "en",
+        lower_case: bool = True,
+        descape: bool = False,
+        verbose: bool = False,
+        over_write: bool = False,
+    ):
+        self.spm_model = spm_model
+        self.lang = lang
+        self.lower_case = lower_case
+        self.descape = descape
+        self.verbose = verbose
+        self.over_write = over_write
+
+        assert spm_model.exists(), f"spm model file: {spm_model} does not exist"
+        self.moses_punct_normalizer = MosesPunctNormalizer(self.lang)
```

Comment:
> 6) Get parity with existing embedding inference: would be really nice if by the time we merge this we have parity (i.e. resolving the divergence between sacremoses and the moses perl scripts).

Ok, let's prioritize updating `sacremosed` before we push this to `main`. 
However, if this is slow for some external reason, I think it is allright to temporarily sacrifice the parity (if I understand correctly, the processing is different only for texts with badly-formed punctuation, so most inputs won't be affected anyway. Is this correct?)

Reply:
Unfortunately it's not with inputs with badly formed punctuation. The "bug" I noticed was that actually the perl scripts seem to produce badly formed outputs, while `sacremoses` seems to have fixed this. Indeed I think this shouldn't be a blocker, but would be nice to have parity by the time we merge. Siddharth is already looking into the issue.

Reply:
Now with sacremoses>=0.1.0, we can pass the perl compatibility flag here!